src/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_IncrementalAnalyzer.cs

```diff
@@ -50,16 +52,20 @@ public async Task<ImmutableArray<DiagnosticData>> ForceAnalyzeProjectAsync(Proje
             {
                 throw ExceptionUtilities.Unreachable();
             }
-        }
 
-        /// <summary>
-        /// Return list of <see cref="StateSet"/> to be used for full solution analysis.
-        /// </summary>
-        private ImmutableArray<StateSet> GetStateSetsForFullSolutionAnalysis(ImmutableArray<StateSet> stateSets, Project project)
-        {
-            // Include only analyzers we want to run for full solution analysis.
-            // Analyzers not included here will never be saved because result is unknown.
-            return stateSets.WhereAsArray(static (s, arg) => arg.self.IsCandidateForFullSolutionAnalysis(s.Analyzer, s.IsHostAnalyzer, arg.project), (self: this, project));
+            async Task<(ImmutableArray<StateSet> stateSets, ImmutableDictionary<DiagnosticAnalyzer, DiagnosticAnalysisResult> diagnosticAnalysisResults)> ComputeForceAnalyzeProjectAsync()
+            {
+                var allStateSets = await _stateManager.GetOrCreateStateSetsAsync(project, cancellationToken).ConfigureAwait(false);
+                var fullSolutionAnalysisStateSets = allStateSets.WhereAsArray(
+                    static (stateSet, arg) => arg.self.IsCandidateForFullSolutionAnalysis(stateSet.Analyzer, stateSet.IsHostAnalyzer, arg.project),
+                    (self: this, project));
+
+                var compilationWithAnalyzers = await GetOrCreateCompilationWithAnalyzersAsync(
```

Comment:
calls GetOrCreateCompilationWithAnalyzersAsync instead of CreateCompilationXXX.  We want this to reuse a cached value if applicable.